From f1a4f200acd9a1815d67f4c16799ebe6ca216ec4 Mon Sep 17 00:00:00 2001
From: Liu Xinyun <xinyun.liu@intel.com>
Date: Sun, 14 Jul 2019 11:21:28 +0800
Subject: [PATCH 1457/1548] drivers/hyper_dmabuf: fix dead lock introduced when
 virtio tx ring is full

release the unlocked mutex before exit

Change-Id: I730584d69002e4ab2466100f307ef81f15419904
Tracked-on: OAM-84155
Signed-off-by: Liu Xinyun <xinyun.liu@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/dma-buf/hyper_dmabuf/virtio/hyper_dmabuf_virtio_be_drv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/dma-buf/hyper_dmabuf/virtio/hyper_dmabuf_virtio_be_drv.c b/drivers/dma-buf/hyper_dmabuf/virtio/hyper_dmabuf_virtio_be_drv.c
index 67b9c51de6de..27600a55688a 100644
--- a/drivers/dma-buf/hyper_dmabuf/virtio/hyper_dmabuf_virtio_be_drv.c
+++ b/drivers/dma-buf/hyper_dmabuf/virtio/hyper_dmabuf_virtio_be_drv.c
@@ -486,6 +486,7 @@ static int virtio_be_send_req(int vmid, struct hyper_dmabuf_req *req,
 	}
 
 	if (timeout <= 0) {
+		mutex_unlock(&priv->lock);
 		dev_warn(hy_drv_priv->dev, "Requests ring full\n");
 		return -EBUSY;
 	}
-- 
2.20.1

